pkg/event/testevent/test.go

```diff
@@ -51,7 +51,7 @@ type Query struct {
 	event.Query
 	RunID         types.RunID
 	TestName      string
-	TestStepLabel string
+	TestStepLabel *string
```

Comment:
We need to be able to search by empty string, so we will use `nil`-value as a zero-value (instead of `""`).

Reply:
I don't think we can specifically search by empty string, other events might be associated to empty string as well. Why can't we search by event name? Which is basically what we are already doing for `TargetAcquired`:

```
        targetAcquiredEvents, err := jr.testEvManager.Fetch(
                testevent.QueryJobID(coordinates.JobID),
                testevent.QueryRunID(coordinates.RunID),
                testevent.QueryTestName(coordinates.TestName),
                testevent.QueryEventName(target.EventTargetAcquired),
        )
```

Reply:
Since we want to generate arbitrary events from `TargetManager`-s we cannot search by event name.

Reply:
Yes, I realized this only later during the day. Since eventually we want to be able to query by the source of the emission (e.g. `TargetManager`) wouldn't it be a better to establish that the event needs to have a specific name to be picked up, and then when we fix #123 , we change the query based on name to be based on source, and we don't need to change anything else? (e.g. restore the condition that specifying an empty field in the query is not valid). 

Does it really need to be arbitrary events, for what we need to do here?

Reply:
> Does it really need to be arbitrary events, for what we need to do here?

I just don't like to limit a solution to a particular task instead of covering a generic task :)
But the fact it is a temporary solution makes me think that you right (and we should make that as non-invasive as possible).

> ..., and we don't need to change anything else?

Seems to be so.

---

So, should I remove the pointer?

Reply:
I do agree that long term you don't want to impose any constraint in terms of name (in fact, when I wrote the first comment, I did not really think through the fact that it would be the TM itself emitting the event, which I realized only via a background thread later) :) 

> I believe I should remove the pointer, right? :)
Yes, and would restore the condition that checks for zero value in the query. I believe we should as much as possible translate "this field should be zero value" into some different condition.

Reply:
Removed the pointer and added search by event name. But I was have to introduce other hacks (to make it so-so safe): now if a `TargetManager` tries to emit an event which name is not "`TargerManager`" then `Emit()` returns an error.